examples/timeseries/timeseries_weather_forecasting.py

```diff
@@ -0,0 +1,418 @@
+]
+
+date_time_key = "Date Time"
+
+
+def show_raw_visualization(data):
+    time_data = data[date_time_key]
+    fig, axes = plt.subplots(
+        nrows=7, ncols=2, figsize=(15, 20), dpi=80, facecolor="w", edgecolor="k"
+    )
+    for i in range(len(feature_keys)):
+        key = feature_keys[i]
+        c = colors[i % (len(colors))]
+        t_data = data[key]
+        t_data.index = time_data
+        t_data.head()
+        ax = t_data.plot(
+            ax=axes[i // 2, i % 2],
+            color=c,
+            title="{} - {}".format(titles[i], key),
+            rot=25,
+        )
+        ax.legend([titles[i]])
+    plt.tight_layout()
+
+
+show_raw_visualization(df)
+
+"""
+This heat map shows the correlation between different features.
+"""
+
+corr = df.corr()
+sns.heatmap(corr)
+
+"""
+The input data will include pressure, temperature (in Celsius) and specific humidity.
+The below line graphs show each feature graphed by hour, month and year.
+"""
+
+
+def show_time_based_visualizations(data, idx):
+    selected_feature = feature_keys[idx]
+    time_data = data[date_time_key]
+    fig, axes = plt.subplots(
+        nrows=1, ncols=3, figsize=(15, 5), dpi=80, facecolor="w", edgecolor="k"
+    )
+
+    dataframe = data.copy()
+    dataframe.index = pd.to_datetime(data[date_time_key])
+
+    x_data = [dataframe.index.hour, dataframe.index.month, dataframe.index.year]
+    x_labels = ["Hour of Day", "Month", "Year"]
+
+    for i in range(3):
+        selected_ax = axes[i]
+        sns.lineplot(x=x_data[i], y=dataframe[selected_feature], ax=selected_ax)
+        selected_ax.set(xlabel=x_labels[i], ylabel=titles[idx])
+    plt.tight_layout()
+    plt.show()
+    return
+
+
+# plot pressure
+show_time_based_visualizations(df, 0)
+# plot temperature
+show_time_based_visualizations(df, 1)
+# plot specific humidity
+show_time_based_visualizations(df, 8)
+
+"""
+## Data Preprocessing
+
+Here we are picking 300000 data points for training. The observation is recorded every
+10 mins, that means 6 times in an hour. Thus STEP is equal to 6. We are tracking data
+from past 720 timestamps(i.e. 720/6=120 hours) and this will be used to predict the
+temperature after 72 timestamps(i.e. 76/6=12 hours).
+
+Raw data is normalized using a z score formula. Since every feature has values with
+varying ranges, normalization is done to confine the values in a range of [0,1] before
+training a neural network.
+It is done by subtracting the mean and dividing by the standard deviation of each feature
+"""
+
+train_split = 300000
+past = 720
+future = 72
+step = 6
+learning_rate = 0.005
+validation_steps = 50
+
+batch_size = 256
+buffer_size = 10000
+evaluation_interval = 200
+epochs = 10
+
+
+def normalize(data, train_split):
+    data_mean = data[:train_split].mean(axis=0)
+    data_std = data[:train_split].std(axis=0)
+    return (data - data_mean) / data_std
+
+
+"""
+We will be using selecting few parameters from the dataset to avoid complete memory(RAM)
```

Comment:
I am confused here. The entire dataset is under 50MB. This example is *not* memory intensive at all. It is extremely lightweight.